File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_FriendGamble.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_FriendGamble implements ITableOperate {
 public final static String TAB_FILE_DATA = "/friend_gamble.txt";
 public static final int ID_STAR = 0 ;
public static final int ID_CARDID = 1 ;
public static final int ID_PRO = 2 ;
public static final int MAX_RECORD = 3 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private int m_CardID;
 public int GetCardID() { return m_CardID;}

private int m_Pro;
 public int GetPro() { return m_Pro;}

private int m_Star;
 public int GetStar() { return m_Star;}

public boolean LoadTable(HashMap<Integer, Table_FriendGamble> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_FriendGamble> _hash = (HashMap<Integer, Table_FriendGamble> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_FriendGamble _values = new Table_FriendGamble();
 _values.m_CardID =  Integer.parseInt((String)valuesList.get(ID_CARDID));
_values.m_Pro =  Integer.parseInt((String)valuesList.get(ID_PRO));
_values.m_Star =  Integer.parseInt((String)valuesList.get(ID_STAR));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

File: TLBB1.5/tabletools/CodeTable/CSharp/Table_FriendGamble.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_FriendGamble : ITableOperate
{ private const string TAB_FILE_DATA = "friend_gamble.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_STAR,
ID_CARDID,
ID_PRO,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private int m_CardID;
 public int CardID { get{ return m_CardID;}}

private int m_Pro;
 public int Pro { get{ return m_Pro;}}

private int m_Star;
 public int Star { get{ return m_Star;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_FriendGamble _values = new Tab_FriendGamble();
 _values.m_CardID =  Convert.ToInt32(valuesList[(int)_ID.ID_CARDID] as string);
_values.m_Pro =  Convert.ToInt32(valuesList[(int)_ID.ID_PRO] as string);
_values.m_Star =  Convert.ToInt32(valuesList[(int)_ID.ID_STAR] as string);

 _hash.Add(nKey,_values); }


}
}

